### ISSUE TITLE
E2E_TAGS with spaces prevents sourcing .env file

### ISSUE THREAD

**Issue by thbkrkr** [elastic/cloud-on-k8s#6596]
e2e tests have Go build tags in place to give the ability to run a subsets of the tests. We use them for `eks-arm`:

https://github.com/elastic/cloud-on-k8s/blob/d33840d0d87d3d81d2e9098dfd1c92b94499a8e0/.buildkite/e2e/nightly-main-matrix.yaml#L54-L58

It is combined in the Makefile and use by the go tests runner:

```Makefile
# combine e2e tags (es, kb, apm etc.)  with go tags (release) to ensure test code that imports generated code works
# this relies on the deprecated space separated build constraints in Go which makes construction in make easier
E2E_TAGS += $(GO_TAGS)
export E2E_TAGS

e2e-run: go-generate
	go run -tags='$(GO_TAGS)' test/e2e/cmd/main.go run \
		--operator-image=$(OPERATOR_IMAGE) \
		--e2e-image=$(E2E_IMG) \
		--e2e-tags='$(E2E_TAGS)' \
...
``` 

This has caused problems in the past: https://github.com/elastic/cloud-on-k8s/issues/6299#issuecomment-1376979301.

This causes an indirect issue since I introduced the script `set-deployer-config.sh` that `source .env` because of the spaces in the val value.
```sh
make run-deployer
/buildkite/builds/bk-agent-elastic-k8s-167966427351480
0400/elastic/cloud-on-k8s-operator-nightly/.buildkite/scripts/test/../../../.env: line 6: kb: command not found
```

To reproduce:
```
echo "E2E_TAGS=es kb apm ent beat agent" >> .env
.buildkite/scripts/test/set-deployer-config.sh
```

It seemed to me that writing an env var value without double quotes isn't valid so I opened a PR to double quotes the value but then it doesn't work well with what is done by the Makefile. It will render: `--e2e-tags='"es kb apm ent beat agent" release'`.

A simple fix for this specific bug can be to remove the `source .env` from the `set-deployer-config.sh` script. This script depends only on the two environments variables that are now also set in the environment of the command via the pipeline generator:
https://github.com/elastic/cloud-on-k8s/blob/d33840d0d87d3d81d2e9098dfd1c92b94499a8e0/.buildkite/e2e/pipeline-gen/pipeline.tpl.yaml#L14-L17

I wonder what the real correct solution to handle this.

We could switch to a comma separated build tags to avoid to have this issue with the spaces, especially since the space separated build constraints are deprecated.
Then there is an issue with the pipeline generator when used with flags because the comma is already used to separate the `k=v`. I guess we could accept to support only one E2E_TAG in this mode.
```sh
pipeline-gen -f p=eks-arm,E2E_TAGS="es kb" # could work
pipeline-gen/pipeline-gen -f p=eks-arm,E2E_TAGS=es,kb # problem
```

**Comment by thbkrkr** [comment]
> Then there is an issue with the pipeline generator when used with flags because the comma is already used to separate the k=v. I guess we could accept to support only one E2E_TAG in this mode.

To circumvent this, Peter suggested to simply use a different separator than the comma in the pipeline-gen (https://github.com/elastic/cloud-on-k8s/pull/6587#pullrequestreview-13597516).

**Comment by thbkrkr** [comment]
This is closed by #6587. I created #6661 for the little follow-up.
